src/dreyfus_index.erl

```diff
@@ -157,9 +157,27 @@ handle_call(info, _From, State) -> % obsolete
     Reply = info_int(State#state.index_pid),
     {reply, Reply, State}.
 
+handle_cast({ddoc_updated, DDocResult},
+            #state{
+                index=#index{sig=Sig, dbname=DbName}
+            }=State) ->
+    case check_if_index_is_deleted(DbName, DDocResult, Sig) of
+        true ->
+            erlang:send_after(index_shutdown_delay(), self(), index_deleted),
```

Comment:
@rnewson  what is the way to "let any requests to it finish"?  How do we know that requests are finished? check if `Waiters` == [] will not work for stale requests that don't wait for indexing to catch up.